airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -602,22 +609,50 @@ def execute(self, context: Context) -> dict:
                 raise
             self.log.info("Cluster already exists.")
             cluster = self._get_cluster(hook)
-
-        # Check if cluster is not in ERROR state
-        self._handle_error_state(hook, cluster)
-        if cluster.status.state == cluster.status.State.CREATING:
-            # Wait for cluster to be created
-            cluster = self._wait_for_cluster_in_creating_state(hook)
-            self._handle_error_state(hook, cluster)
-        elif cluster.status.state == cluster.status.State.DELETING:
-            # Wait for cluster to be deleted
-            self._wait_for_cluster_in_deleting_state(hook)
-            # Create new cluster
-            cluster = self._create_cluster(hook)
+        if not self.deferrable:
+            # Check if cluster is not in ERROR state
             self._handle_error_state(hook, cluster)
-
+            if cluster.status.state == cluster.status.State.CREATING:
+                # Wait for cluster to be created
+                cluster = self._wait_for_cluster_in_creating_state(hook)
+                self._handle_error_state(hook, cluster)
+            elif cluster.status.state == cluster.status.State.DELETING:
+                # Wait for cluster to be deleted
+                self._wait_for_cluster_in_deleting_state(hook)
+                # Create new cluster
+                cluster = self._create_cluster(hook)
```

Comment:
In case of async operator _create_cluster can be a async API call here

Reply:
fix